x-pack/plugins/security_solution/public/management/services/policies/hooks.ts

```diff
@@ -40,3 +45,35 @@ export function useGetEndpointSpecificPolicies(
     }
   );
 }
+
+export function useGetAgentCountForPolicy({
+  onError,
+  policyIds,
+}: {
+  onError?: (error: ServerApiError) => void;
+  policyIds: string[];
+}): QueryObserverResult<GetAgentPoliciesResponse> {
+  // is there a more elegant way to do this? w/o the conditional, the call is a bad request when the policyId is empty
+  const http = useHttp();
+  return useQuery<GetAgentPoliciesResponse, ServerApiError>(
+    ['endpointCountForPolicy', policyIds],
+    () => {
+      if (policyIds.length > 0) {
+        return sendGetAgentPolicyList(http, {
+          query: {
+            perPage: 50,
+            kuery: `${AGENT_POLICY_SAVED_OBJECT_TYPE}.package_policies: (${policyIds.join(
+              ' or '
+            )})`,
+          },
+        });
+      }
+      return sendGetAgentPolicyList(http, { query: {} });
+    },
+    {
+      refetchIntervalInBackground: false,
+      refetchOnWindowFocus: false,
+      onError,
+    }
+  );
+}
```

Comment:
You can do something like this instead:

```suggestion
export function useGetAgentCountForPolicy({
  policyIds,
}: {
  policyIds: string[];
}): QueryObserverResult<GetAgentPoliciesResponse> {
  const http = useHttp();
  const toasts = useToasts();
  return useQuery<GetAgentPoliciesResponse, ServerApiError>(
    ['endpointCountForPolicy', policyIds],
    () => {
      return sendGetAgentPolicyList(http, {
        query: {
          perPage: 50,
          kuery: `${AGENT_POLICY_SAVED_OBJECT_TYPE}.package_policies: (${policyIds.join(' or ')})`,
        },
      });
    },
    {
      refetchIntervalInBackground: false,
      refetchOnWindowFocus: false,
      onError: (error: ServerApiError) => {
        toasts.addError(error as unknown as Error, {
          title: i18n.translate('xpack.securitySolution.endpoint.policiesList.errorToastTitle', {
            defaultMessage: 'Empty policy Ids! Can not fetch policy list with no policy IDs',
          }),
        });
      },
    }
  );
}
```